app/post/postDetailsDirective.js

```diff
@@ -332,6 +334,33 @@
             }
             return text;
         }
+
+        postDetailsCtrl.pdfDialog = function(ev, pdf) {
+            var readablePdf = {url: ''};
+            PdfService.getReadableURL(pdf.url, setPdfURL, readablePdf).then(
```

Comment:
Não seria melhor essa funação retornar a url do pdf quando a promisse for resolvida não?